### PR TITLE
Facet

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/compile/mark/mark.ts [hunk 1]
```diff
@@ -79,12 +90,10 @@ function parsePathMark(model: UnitModel) { // TODO: extract this into compilePat
...
       type: 'group',
-      from: extend(
-        // If has facet, `from.data` will be added in the cell group.
-        // Otherwise, add it here.
-        isFaceted ? {} : dataFrom,
-        {transform: transform}
-      ),
+      from: {
+        data: dataFrom(model),
+        transform: transform
```
Comment (domoritz) [inline src/compile/mark/mark.ts]: I thought transform is not possible any more. Or did you just leave this in here for later?

File: src/compile/mark/mark.ts [hunk 2]
```diff
@@ -100,51 +109,35 @@ function parsePathMark(model: UnitModel) { // TODO: extract this into compilePat
...
+  marks.push({
+    name: model.name('marks'),
+    type: markCompiler[mark].markType(),
+    from: extend(
+      {data: dataFrom(model)},
+      model.stack() ? // Stacked Chart need stack transform
+        { transform: stackTransforms(model, false) } :
+      model.has(ORDER) ?
+        // if non-stacked, detail field determines the layer order of each mark
+        { transform: [{type:'sort', by: sortBy(model)}] } :
```
Comment (domoritz) [inline src/compile/mark/mark.ts]: Need to move this out as well

File: src/compile/mark/mark.ts (hunk 1 above)
Comment (kanitw) [inline src/compile/mark/mark.ts]: that's your job!

Comment (kanitw) [inline src/compile/mark/mark.ts]: (That's why I need to hurry to get this facet in, so you can do your job...)

Comment (domoritz) [inline src/compile/mark/mark.ts]: Ahh, now I get what your plan was. I was a bit confused by this, tbh.

**Pull request merged**